### DISCUSSION TITLE
find API for a function (tickFormat as an example)

### DISCUSSION THREAD

**Discussion by langenhoff** [observablehq/plot#2171] [Q&A] [answered]
Using the plot documention, I searched for *tickFormat* in the search bar. One match is the section about scales, which provides this hint: "The tickFormat option can either be a [d3.format](https://d3js.org/d3-format) string or a function that takes a tick value and returns the corresponding string". 

Some examples from the gallery and code snippets I've seen lead to the assumption that it possible to have additional arguments, like so: 

     tickFormat: (d, i, _) => ...

I guess: d is the current data, i is the index and the underscore represents the array of values. But this is only a guess.

What is the recommended way of looking up the API (especially the arguments)?

**Comment by Fil** [comment, marked as answer]
Plot includes TypeScript declarations with extensive documentation. We highly recommend using an editor with enhanced code completion such as Visual Studio Code or Observable.

In this particular case we have:

```
  /**
   * How to format inputs (abstract values) for axis tick labels; one of:
   *
   * - a [d3-format][1] string for numeric scales
   * - a [d3-time-format][2] string for temporal scales
   * - a function passed a tick *value* and *index*, returning a string
   *
   * [1]: https://d3js.org/d3-time
   * [2]: https://d3js.org/d3-time-format
   */
  tickFormat?: string | ((d: any, i: number) => any) | null;
```

This means that the `_` (which is indeed the array of values) is an implementation detail which is not part of the official API, and not guaranteed to stay.

**Reply by langenhoff** [reply]
Understood. Thanks for your answer. Your code snippet is located in *legends.d.ts*, right? I use Visual Studio Code for observable plot (but without any build step, just plain html/javascript with <script> tags). What is necessary to make the TypeScript declarations visible to Visual Studio Code?

**Reply by Fil** [reply]
“[Intellisense](https://code.visualstudio.com/docs/languages/typescript#_intellisense)” is supposed to work out of the box when you use npm or yarn. I'm not sure how to help you fix it.

**Reply by langenhoff** [reply]
Thank you, I think I can figure it out myself. I'm just not very familiar with this tool chain.
